api/controllers/CampaignBotController.js

```diff
@@ -22,433 +17,390 @@ const CMD_REPORTBACK = (process.env.GAMBIT_CMD_REPORTBACK || 'P');
-    
-    if (promptUser) {
-      return this.sendMessage(req, res, this.bot.msg_ask_quantity);
-    }
-
-    const quantity = req.incoming_message;
-    // @todo: Extract any numbers we can find instead of invalidating input based
-    // on whether it contains any words (could contain noun, verb, or 'around 90')
-    if (helpers.hasLetters(quantity) || !parseInt(quantity)) {
-      return this.sendMessage(req, res, this.bot.msg_invalid_quantity);
-    }
-
-    this.reportbackSubmission.quantity = parseInt(quantity);
-
-    return this.reportbackSubmission
-      .save()
-      .then(() => {
-        this.debug(req, `saved quantity:${this.reportbackSubmission.quantity}`);
-        return this.collectPhoto(req, res, true);
-      });
-  }
+  /**
+   * Handles conversation for collecting ReportbackSubmission data or posting.
+   * @param {object} req - Express request
+   */
+  continueReportbackSubmission(req) {
+    this.debug(req, 'continueReportbackSubmission');
 
-  collectWhyParticipated(req, res, promptUser) {
-    this.debug(req, `collectWhyParticipated prompt:${promptUser}`);
+    const submission = req.signup.draft_reportback_submission;
+    const ask = ( req.query.start || false );
 
-    if (promptUser) {
-      return this.sendMessage(req, res, this.bot.msg_ask_why_participated);
+    if (!submission.quantity) {
+      return this.collectReportbackProperty(req, 'quantity', ask);
+    }
+    if (!submission.photo) {
+      return this.collectReportbackProperty(req, 'photo', ask);
+    } 
+    if (!submission.caption) {
+      return this.collectReportbackProperty(req, 'caption', ask);
+    }
+    if (!submission.why_participated) {
+      return this.collectReportbackProperty(req, 'why_participated', ask);
     }
 
-    this.reportbackSubmission.why_participated = req.incoming_message;
-
-    return this.reportbackSubmission
-      .save()
-      .then(() => {
-        this.debug(req, `saved why_participated:${req.incoming_message}`);
-        return this.postReportback(req, res);
-      });
+    // If we're here, we have a completed submission but the POST request
+    // likely failed.
+    // TODO: Check failed_at before sending? Message about whoops trying again?
+    logger.warn('no messages sent from continueReportbackSubmission');
+    return this.postReportback(req);
   }
 
-  createReportbackSubmission(req, res) {
+  /**
+   * Creates new ReportbackSubmission model and updates Signup model's draft.
+   * @param {object} req - Express request
+   * @return {string} - Message to send and begin collecting Reportback data.
+   */
+  createReportbackSubmission(req) {
     this.debug(req, 'createReportbackSubmission');
 
-    this.reportbackSubmission = new dbRbSubmissions({
-      campaign: this.campaignId,
-      user: req.user_id, 
-    });
+    return app.locals.db.reportbackSubmissions
+      .create({
+        campaign: req.campaign_id,
+        user: req.user_id, 
+      })
+      .then(reportbackSubmission => {
```

Comment:
@deadlybutter It's the code style. On other projects we use [eslint](https://github.com/DoSomething/eslint-config) @DFurnes created. It's based on AirBnb JS code style: https://github.com/airbnb/javascript, which requires the parentheses:

> 8.4 If your function takes a single argument and doesn’t use braces, omit the parentheses. Otherwise, always include parentheses around arguments. eslint: arrow-parens jscs: disallowParenthesesAroundArrowParam

> Why? Less visual clutter.

> https://github.com/airbnb/javascript#arrows--one-arg-parens